framework/src/problems/FEProblemBase.C

```diff
@@ -6638,13 +6638,19 @@ FEProblemBase::checkNonlinearConvergence(std::string & msg,
           << '\n';
       reason = MooseNonlinearConvergenceReason::CONVERGED_SNORM_RELATIVE;
     }
-    else if (divtol > 0 && fnorm > the_residual * divtol)
+    else if (divtol > 0 && it >= _nl_forced_its && fnorm > the_residual * divtol)
```

Comment:
We have an existing parameter `force_iteration`. Could we just use that?

Reply:
force_iteration is PETSC boolean. 
so i can define a method in FEProbelmBase to check if a specific iteration is forced or not, and use that method in `PetscSupport.C` to set force_iteration = True
those this sounds good?

Reply:
I see the point here. We should use `_nl_forced_its` instead of `force_iteration` in the most of the place.

`force_iteration` is triggered by PETSc option `-snes_force_iteration`.  `force_iteration` should only appear in `petscNonlinearConverged`.


1) if `force_iteration` is true, and `_nl_forced_its` is zero, and we should set `_nl_forced_its` to `1 `.

2) If `force_iteration` is false, but `_nl_forced_its` is larger than zero, we should call SNESSetForceIteration(snes, PETSC_TRUE)

3) if `force_iteration` is false, and `_nl_forced_its` is zero, do nothing

4) if `force_iteration` is true, and `_nl_forced_its` is larger than zero, do nothing

Reply:
I implemented  the above logic in `FEProblemBase:: setNLForcedIterations `.